core/src/main/java/feast/core/grpc/CoreServiceImpl.java

```diff
@@ -113,6 +95,21 @@ public void listFeatureSets(
     }
   }
 
+  @Override
+  public void getFeatureStatistics(
+      GetFeatureStatisticsRequest request,
+      StreamObserver<GetFeatureStatisticsResponse> responseObserver) {
+    try {
+      GetFeatureStatisticsResponse response = statsService.getFeatureStatistics(request);
+      responseObserver.onNext(response);
+      responseObserver.onCompleted();
+    } catch (RetrievalException | IllegalArgumentException | IOException e) {
+      log.error("Exception has occurred in GetFeatureStatistics method: ", e);
+      responseObserver.onError(
+          Status.INTERNAL.withDescription(e.getMessage()).withCause(e).asRuntimeException());
+    }
```

Comment:
Would it make sense to also return the message to the end user of other exceptions? It's obviously not a good practice long term, but for the time being it would make debugging easier.